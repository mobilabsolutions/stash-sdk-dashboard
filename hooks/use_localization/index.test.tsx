/* eslint-env jest */

import { useLocalization } from './index'
import { testRender } from '../../test_utils'
import localizedAmount from './localized_amount'

const PaymentDashboard = () => {
  const { getText } = useLocalization()

  return <span>{getText('Payment Dashboard')}</span>
}

it('Should "getText" in "en"', () => testRender(PaymentDashboard))
it('Should "getText" in "de"', () =>
  testRender(PaymentDashboard, null, { locale: 'de' }))

const Number = () => {
  const { formatNumber } = useLocalization()

  return <span>{formatNumber(123.45)}</span>
}

it('Should "formatNumber" in "en"', () => testRender(Number))
it('Should "formatNumber" in "de"', () =>
  testRender(Number, null, { locale: 'de' }))

const Amount = () => {
  const { formatAmount } = useLocalization()

  return <span>{formatAmount('EUR', 12345).value}</span>
}
it('Should "formatAmount" in "en"', () => testRender(Amount))
it('Should "formatAmount" in "de"', () =>
  testRender(Amount, null, { locale: 'de' }))

it('"localizedAmount" should return all properties for a currency', () => {
  expect(localizedAmount('EUR', 12345, 'de')).toMatchSnapshot()
  expect(localizedAmount('EUR', 12345, 'en')).toMatchSnapshot()
})
