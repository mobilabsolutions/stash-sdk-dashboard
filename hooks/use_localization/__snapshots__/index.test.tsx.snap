// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`"localizedAmount" should return all properties for a currency 1`] = `
Object {
  "decimal": ",",
  "group": ".",
  "symbol": "€",
  "symbolAtEnd": true,
  "value": "12.345,00 €",
}
`;

exports[`"localizedAmount" should return all properties for a currency 2`] = `
Object {
  "decimal": ".",
  "group": ",",
  "symbol": "€",
  "symbolAtEnd": false,
  "value": "€12,345.00",
}
`;

exports[`Should "formatAmount" in "de" 1`] = `
<span>
  12.345,00 €
</span>
`;

exports[`Should "formatAmount" in "en" 1`] = `
<span>
  €12,345.00
</span>
`;

exports[`Should "formatNumber" in "de" 1`] = `
<span>
  123,45
</span>
`;

exports[`Should "formatNumber" in "en" 1`] = `
<span>
  123.45
</span>
`;

exports[`Should "getText" in "de" 1`] = `
<span>
  Zahlungs-Dashboard
</span>
`;

exports[`Should "getText" in "en" 1`] = `
<span>
  Payment Dashboard
</span>
`;
